### PR TITLE
Implement basic rl agents

### PR DESCRIPTION
**Pull request by SuperKoopa2109**

### CODE REVIEW

File: IRL_agents.py [hunk 1]
```diff
@@ -74,6 +74,21 @@ def compute_value_estimate(self, trajs: List[List[Any]]) -> List[float]:
...
+
+            for traj in trajs:
+                for i, state in enumerate(traj):
+                    temp_value += self.gamma**i * self._approx_func(x=state, center=approx_center)
+
+            temp_value /= len(trajs)
+            action_value_estimates.append(temp_value)
+
+        return action_value_estimates
+
```
Comment (maxserra) [inline IRL_agents.py]: How is this function any different from the one above?

File: GridWorld_environments.py [hunk 2]
```diff
@@ -36,10 +37,30 @@ def __init__(self,
...
+            #    self.int_goals.append((np.random.choice(range(size[0])), np.random.choice(range(size[1]))))
+            
+            #get random goal -- for now just one goal
+            self.goals = []
+            self.goals.append((np.random.choice(range(size[0])), np.random.choice(range(size[1]))))
+        else:
+            self.walls = walls
+            self.traps = traps
+            self.int_goals = intermediate_goals
+            self.goals = goals
```
Comment (maxserra) [inline GridWorld_environments.py]: Maybe we pass the % of walls, traps, int_goals and goals as a list/dict? or with additional arguments? So we can be flexible?
Right now, if the size is 4x4, there would be no traps nor anything :/

File: RL_arena.py [hunk 3]
```diff
@@ -86,46 +151,92 @@ def irl_reward_estimation(env: Grid_World, optimal_trajectories: List[List[Any]]
...
 
 if __name__ == "__main__":
 
     parser = argparse.ArgumentParser()
 
     parser.add_argument("-vi", "--value-iteration", required=False, default=False, action="store_true")
+    parser.add_argument("-ql", "--q-learning", required=False, default=False, action="store_true")
     parser.add_argument("-gt", "--generate-trajectories", required=False, default=False, action="store_true")
     parser.add_argument("-irl", "--inverse-rl", required=False, default=False, action="store_true")
+    parser.add_argument("-plt", "--plots", required=False, default=False, action="store_true")
```
Comment (maxserra) [inline RL_arena.py]: Is this argument used? I think it should be used, not removed :)

File: GridWorld_environments.py [hunk 4]
```diff
@@ -26,7 +26,8 @@ def __init__(self,
                  walls: List[Tuple[int]] = [],
```
Comment (SuperKoopa2109) [inline GridWorld_environments.py:18]: This is a test comment @maxserra

File: GridWorld_environments.py [hunk 5]
```diff
@@ -26,7 +26,8 @@ def __init__(self,
...
                  traps: List[Tuple[int]] = [],
                  intermediate_goals: List[Tuple[int]] = [],
-                 goals: List[Tuple[int]] = []):
+                 goals: List[Tuple[int]] = [],
+                 randomize_board: bool = False):
 
         self.possible_actions = [(1, 0), (0, 1), (-1, 0), (0, -1)]  # South, East, North, West
```
Comment (SuperKoopa2109) [inline GridWorld_environments.py]: A test review.

**Review by maxserra — CHANGES_REQUESTED** [review]

File: RL_arena.py [hunk 6]
```diff
@@ -89,43 +142,88 @@ def irl_reward_estimation(env: Grid_World, optimal_trajectories: List[List[Any]]
 
         # step 6: find optimal policy under new reward function and add to 'candidate_policies' list
         env.set_reward_func(reward_func)
-        candidate_policies.append(train_value_iteration(gw_env=env))
+        # minmax_scaler = MinMaxScaler()
+        print('env.get_board \n', env.get_board())
+        reward_func_preds.append(np_normalize(abs(deepcopy(env.get_board()))))
```
Comment (maxserra) [inline RL_arena.py]: @SuperKoopa2109 why the `abs()` function??

**Review by maxserra — APPROVED** [review]

**Pull request merged**